### PR TITLE
doc: ap-protect: update to user guide and glossary

### PR DESCRIPTION
**Pull request by greg-fer**
Updated the user guide about AP-Protect with additional info requested by Tech Support. Also updated the glossary term with a link to the user guide and clarified term usage. NCSDK-22122.

### CODE REVIEW

File: doc/nrf/app_dev/ap_protect/index.rst [hunk 1]
```diff
@@ -118,19 +118,20 @@ Based on the available implementation types, you can configure the access port p
...
    * - Locked
      - :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` (:kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` for Secure AP-Protect)
-     - In this state, CPU writes to enable and lock AP-Protect. UICR is not modified.
+     - In this state, CPU uses the MDK system start-up file to write to enable and lock AP-Protect. UICR is not modified.
      - Hardware and software
    * - Authenticated
      - :kconfig:option:`CONFIG_NRF_APPROTECT_USER_HANDLING` (:kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USER_HANDLING` for Secure AP-Protect)
-     - In this state, AP-Protect is left enabled and it is up to the user handler to unlock the device if needed.
+     - In this state, AP-Protect is left enabled and it is up to the user-space code to handle unlocking the device if needed.
+       The MDK will close the application subsystem access port AHB-AP, but not lock it, so the AHB-AP can be reopened via UART or CTRL-AP.
```
Comment (simtind) [inline doc/nrf/app_dev/ap_protect/index.rst:130]:
```suggestion
       The MDK will close the application subsystem access port AHB-AP, but not lock it, so the AHB-AP can be reopened by the fw. 
       Reopening the AHB-AP would typically be preceded by a handshake operation over UART, CTRL-AP Mailboxes or some other communication channel.
```

The FW reopens the access port, not the CTRL-AP / UART directly. The FW should receive and verify the reopening request through some communication channel like UART, CTRL-AP Mailboxes (available on nrf53 only (nrf91 maybe?)) and make sure the request is valid by requiring a cryptographic handshake of some kind, usually asking the requester to decrypt a random piece of information and pass it back for verification.

**Review by hellesvik-nordic — APPROVED** [review]

File: doc/nrf/app_dev/ap_protect/index.rst [hunk 2]
```diff
@@ -118,19 +118,21 @@ Based on the available implementation types, you can configure the access port p
 
    * - AP-Protect state
      - Related Kconfig option in the |NCS|
-     - Description
+     - Description of the AP-Protect state
      - AP-Protect implementation type
    * - Locked
      - :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` (:kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` for Secure AP-Protect)
-     - In this state, CPU writes to enable and lock AP-Protect. UICR is not modified.
+     - In this state, CPU uses the MDK system start-up file to write to enable and lock AP-Protect. UICR is not modified.
```
Comment (simtind) [inline doc/nrf/app_dev/ap_protect/index.rst:125]:
```suggestion
     - In this state, CPU uses the MDK system start-up file to enable and lock AP-Protect. UICR is not modified.
```

Not sure write to is neccessary here, and it makes the sentence a bit clunky

File: doc/nrf/app_dev/ap_protect/index.rst [hunk 3]
```diff
@@ -118,19 +118,21 @@ Based on the available implementation types, you can configure the access port p
...
      - Hardware and software
    * - Authenticated
      - :kconfig:option:`CONFIG_NRF_APPROTECT_USER_HANDLING` (:kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_USER_HANDLING` for Secure AP-Protect)
-     - In this state, AP-Protect is left enabled and it is up to the user handler to unlock the device if needed.
+     - In this state, AP-Protect is left enabled and it is up to the user-space code to handle unlocking the device if needed.
+       The MDK will close the application subsystem access port AHB-AP, but not lock it, so the AHB-AP can be reopened by the firmware.
```
Comment (simtind) [inline doc/nrf/app_dev/ap_protect/index.rst:130]:
```suggestion
       The MDK will close the debug AHB-AP, but not lock it, so the AHB-AP can be reopened by the firmware.
```
This is per domain, so shouldn't mention application subsystem here (network will operate independently from application core for example)

May be that AHB-AP is not sufficiently described at this point, maybe we want to add a description further up if we have a concepts section. Alternatively we can drop AHB-AP and just talk about debugging being closed or something.

File: doc/nrf/glossary.rst [hunk 4]
```diff
@@ -612,9 +614,10 @@ Glossary
       It comes with digital encoder and decoder for transferring audio data.
       It is being gradually supplanted by the :term:`LC3 codec` developed for :term:`LE Audio`.
 
-   Secure Access Port Protection (SECUREAPPROTECT)
-      A register used to prevent read and write access to all secure CPU registers and secure memory-mapped addresses.
-      See :term:`Access Port Protection (APPROTECT)`.
+   Secure access port protection mechanism (Secure AP-Protect)
+      A mechanism used to prevent read and write access to all secure CPU registers and secure memory-mapped addresses.
+      In the |NCS|, it uses ``SECUREAPPROTECT`` registers and several Kconfig options, as described in :ref:`app_approtect`.
+      See also :term:`Access port protection mechanism (AP-Protect)`.
```
Comment (simtind) [inline doc/nrf/glossary.rst:621]:
```suggestion

AHB Access Port (AHB-AP)
      An Arm Coresight component that allows a debug probe to access CPU cores for programming and debugging.
      Access through an AHB-AP can be blocked with AP-Protect.

```

**Review by simtind — APPROVED** [review]

**Pull request merged**